jt-classbreaks/src/test/java/it/geosolutions/jaiext/classbreaks/ClassBreaksOpImageTest.java

```diff
@@ -204,6 +199,154 @@ public void testNaturalBreaksHistogram() throws Exception {
+        RenderedImage image = createImage();
+
+        ParameterBlockJAI pb = new ParameterBlockJAI(new ClassBreaksDescriptor());
+        pb.addSource(image);
+        pb.setParameter("method", ClassificationMethod.NATURAL_BREAKS);
+        pb.setParameter("numClasses", 4);
+        pb.setParameter("percentages",true);
+        RenderedImage op  = JAI.create("ClassBreaks", pb, null);
+        Classification classification =
+                (Classification) op.getProperty(ClassBreaksDescriptor.CLASSIFICATION_PROPERTY);
+        assertNotNull(classification);
+        double[] percentages = classification.getPercentages();
+        assertEquals(4, percentages.length);
+        assertTrue(percentages[0]==18.75);
+        assertTrue(percentages[1]==43.75);
+        assertTrue(percentages[2]==12.5);
+        assertTrue(percentages[3]==25.0);
+    }
+
+    @Test
+    public void testNaturalBreaksHistogramWithPercentages() throws Exception {
+        RenderedImage image = createImage();
+
+        ParameterBlockJAI pb = new ParameterBlockJAI(new ClassBreaksDescriptor());
+        pb.addSource(image);
+        pb.setParameter("method", ClassificationMethod.NATURAL_BREAKS);
+        pb.setParameter("numClasses", 4);
+        pb.setParameter("extrema", getExtrema(image));
+        pb.setParameter("histogram", true);
+        pb.setParameter("histogramBins", 100);
+        pb.setParameter("percentages",true);
+        RenderedImage op  = JAI.create("ClassBreaks", pb, null);
+        Classification classification =
+                (Classification) op.getProperty(ClassBreaksDescriptor.CLASSIFICATION_PROPERTY);
+        assertNotNull(classification);
+        double[] percentages = classification.getPercentages();
+        assertEquals(4, percentages.length);
+        assertTrue(Math.floor(percentages[0])==16.0);
+        assertTrue(Math.ceil(percentages[1])==42.00);
+        assertTrue(Math.floor(percentages[2])==16.0);
+        assertTrue(percentages[3]==25.0);
+    }
+
+    @Test
+    public void testQuantileBreaksWithPercentages() throws Exception {
+        RenderedImage image = createImage();
+
+        ParameterBlockJAI pb = new ParameterBlockJAI(new ClassBreaksDescriptor());
+        pb.addSource(image);
+        pb.setParameter("method", ClassificationMethod.QUANTILE);
+        pb.setParameter("numClasses", 4);
+        pb.setParameter("percentages",true);
+        RenderedImage op  = JAI.create("ClassBreaks", pb, null);
+        Classification classification =
+                (Classification) op.getProperty(ClassBreaksDescriptor.CLASSIFICATION_PROPERTY);
+        assertNotNull(classification);
+        double[] percentages = classification.getPercentages();
+        assertEquals(percentages.length, 4);
+        assertTrue(percentages[0]== 18.75);
+        assertTrue(percentages[1]== 31.25);
+        assertTrue(percentages[2]== 25.0);
+        assertTrue(percentages[3]== 25.0);
+        // Test required classes > breaks
```

Comment:
One test method, one test, please split and give the method a name that talks about what the test is doing. Do the same for all other tests.

Reply:
fixed